sdk/Trace/TextMapGetterSetter.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Trace;
+
+use function array_keys;
+use ArrayAccess;
+use function get_class;
+use function gettype;
+use InvalidArgumentException;
+use function is_array;
+use function is_object;
+use function is_string;
+use OpenTelemetry\Trace as API;
+use function sprintf;
+use function strtolower;
+
+/**
+ * @see https://github.com/open-telemetry/opentelemetry-specification/blob/v1.6.1/specification/context/api-propagators.md#textmap-propagator Getter and Setter.
+ *
+ * Default implementation of {@see API\PropagationGetter} and {@see API\PropagationSetter}.
+ * This type is used if no custom getter/setter is provided to {@see API\TextMapPropagator::inject()} or {@see API\TextMapPropagator::extract()}.
+ */
+class TextMapGetterSetter implements API\PropagationGetter, API\PropagationSetter
```

Comment:
This used to be called `PropagationMap` but I renamed it to be a bit more clear on what it actually is.

Reply:
Although maybe `ArrayGetterSetter` would be even better...

Reply:
There aren't going to be times when we are GetterSetter'ing anything besides an array, correct?

Reply:
I don't think we can have something for _all_ use cases. The majority of the time the carrier is most likely going to be an array, so having that be the default is fine. If that is not the case for someone, then they would need to define their own getter/setter type. I.e. there is a dedicated object or something.

Might make sense to make the carrier the 2nd argument as that might be more common than using a custom context?

Reply:
Agreed that a carrier will certainly be more common than using a custom context.  Sounds like a plan to me.